src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,33 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
 public class SalaryInfo {
+    private final DateTimeFormatter formatter
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy", Locale.ENGLISH);
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder reportBuilder = new StringBuilder();
+
+        for (String i : names) {
+            int employeeSalarySum = 0;
+            for (String j : data) {
+                String[] splitData = j.split(" ");
+                LocalDate particularDate = LocalDate.parse(splitData[0], formatter);
+                LocalDate convertedDateFrom = LocalDate.parse(dateFrom, formatter);
+                LocalDate convertedDateTO = LocalDate.parse(dateTo, formatter);
+                int compareDateFrom = particularDate.compareTo(convertedDateFrom);
+                int compareDateTo = particularDate.compareTo(convertedDateTO);
+                if (splitData[1].equals(i) && (compareDateFrom >= 0 && compareDateTo <= 0)) {
```

Comment:
![image](https://user-images.githubusercontent.com/12864128/209363641-2ec1e3ef-597b-4fd0-ba01-a41c3c951e2f.png)
Move indexes to constant